### PR TITLE
KYLIN-3839 Strorage clean up after refreshing and deleting segment

### PR DESCRIPTION
**Pull request by Wayne1c**

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by Wayne1c** [comment]
pr updated

File: server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java [hunk 1]
```diff
@@ -550,7 +558,30 @@ public CubeInstance deleteSegment(CubeInstance cube, String segmentName) throws
...
+        CubeInstance cubeInstance = CubeManager.getInstance(getConfig()).updateCubeDropSegments(cube, toDelete);
+
+        cleanSegmentStorage(Collections.singletonList(toDelete));
+
+        return cubeInstance;
+    }
+
+    // clean segment data in hbase and hdfs
+    private void cleanSegmentStorage(List<CubeSegment> toRemoveSegs) throws IOException {
+        if (KylinConfig.getInstanceFromEnv().cleanStorageAfterDelOperation()) {
```
Comment (shaofengshi) [inline server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java]: If not enabled, directly return, to make the code more clear (less embed logic);

File: server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java [hunk 2]
```diff
@@ -550,7 +558,30 @@ public CubeInstance deleteSegment(CubeInstance cube, String segmentName) throws
...
+            if (toRemoveSegs != null) {
+                List<String> toDropHTables = Lists.newArrayListWithCapacity(toRemoveSegs.size());
+                List<String> toDelHDFSPaths = Lists.newArrayListWithCapacity(toRemoveSegs.size());
+                for (CubeSegment seg : toRemoveSegs) {
+                    toDropHTables.add(seg.getStorageLocationIdentifier());
+                    toDelHDFSPaths.add(JobBuilderSupport.getJobWorkingDir(seg.getConfig().getHdfsWorkingDirectory(), seg.getLastBuildJobID()));
+                }
+
+                if (!toRemoveSegs.isEmpty()) {
```
Comment (shaofengshi) [inline server-base/src/main/java/org/apache/kylin/rest/service/CubeService.java]: Please combine this line with the previous " if (toRemoveSegs != null)" check.

**Pull request merged**